### PR TITLE
Support Wal pvc for a given OSD

### PR DESCRIPTION
**Pull request by kesavanvt**
**Description :**

- Rook has added support to configure a wal PVC for an OSD in https://github.com/rook/rook/pull/5580

- This PR is about enabling the same support via OCS

- It can be used by creating a new walPVCTemplate.

A template will look like this:
```
dataPVCTemplate:
    spec:
      # IMPORTANT: Change the storage class depending on your environment
      storageClassName: local-block
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          # Replace the size with the size of localPV prepared
          storage: 500Gi
      volumeMode: Block
walPVCTemplate:
    spec:
      # IMPORTANT: Change the storage class depending on your environment
      storageClassName: local-block-wal
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          # Replace the size with the size of localPV prepared
          storage: 100Gi
      volumeMode: Block

```

### CODE REVIEW

File: pkg/apis/ocs/v1/storagecluster_types.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ type StorageClusterSpec struct {
 	// If enabled, sets the failureDomain to host, allowing devices to be
 	// distributed evenly across all nodes, regardless of distribution in zones
 	// or racks.
-	FlexibleScaling  bool     `json:"flexibleScaling,omitempty"`
+	FlexibleScaling bool `json:"flexibleScaling,omitempty"`
```
Comment (crombus) [inline pkg/apis/ocs/v1/storagecluster_types.go:46]: make this in a separate commit or ignore it.

**Review by crombus — CHANGES_REQUESTED** [review]
rest looks good

File: pkg/apis/ocs/v1/storagecluster_types.go (hunk 1 above)
Comment (kesavanvt) [reply pkg/apis/ocs/v1/storagecluster_types.go:46]: Thanks, have made the above change in a separate commit.

File: pkg/controller/storagecluster/reconcile.go [hunk 2]
```diff
@@ -49,8 +49,10 @@ osd_memory_target_cgroup_limit_ratio = 0.5
 	monCountOverrideEnvVar = "MON_COUNT_OVERRIDE"
 	// EBS represents AWS EBS provisioner for StorageClass
 	EBS StorageClassProvisionerType = "kubernetes.io/aws-ebs"
-	//Name of MetadataPVCTemplate
+	// Name of MetadataPVCTemplate
```
Comment (jarrpa) [inline pkg/controller/storagecluster/reconcile.go:52]: Move to the formatting commit.

**Review by jarrpa — CHANGES_REQUESTED** [review]
Code looks good, commits need some cleanup.

* Rename "code formatting" to "formatting fixes"
* Rename "refactoring existing test" to "use existing PVC templates in tests"

File: pkg/controller/storagecluster/reconcile.go (hunk 2 above)
Comment (kesavanvt) [reply pkg/controller/storagecluster/reconcile.go:52]: Thanks, required changes have been made!

**Review by crombus — APPROVED** [review]

**Review by jarrpa — APPROVED** [review]
/lgtm

**Comment by obnoxxx** [comment]
While this is not officially planned for OCS 4.7, it would still be good to get it in by feature freeze to complement the completed rook work, so people can start playing with it.

**Review by obnoxxx — APPROVED** [review]
/lgtm
